roles/redpanda_broker/tasks/install-certs.yml

```
@@ -38,6 +38,17 @@
     force: "{{ overwrite_certs | default('no') | bool }}"
     mode: "0644"
 
+- name: Install_certs - Copy Node Keys
+  tags:
+    - install_certs
+  ansible.builtin.copy:
+    src: "{{ node_key_file }}"
+    dest: "{{ redpanda_key_file }}"
+    owner: redpanda
+    group: redpanda
+    force: "{{ overwrite_certs | default('no') | bool }}"
+    mode: "0644"
```

```suggestion
- name: Install_certs - Copy Node Keys
  tags:
    - install_certs
  ansible.builtin.copy:
    src: "{{ node_key_file }}"
    dest: "{{ redpanda_key_file }}"
    owner: redpanda
    group: redpanda
    force: "{{ overwrite_certs | default('no') | bool }}"
    mode: "0644"
    when:
    - handle_cert_install | default(false) | bool
    - create_demo_certs | default(false) | bool
```

needs the when block to ensure it will only run in the correct scenario, otherwise looks good

---

So you mean It will only run when handle_cert_install is true and create_demo_certs is true ?

---

@iumesh09 sorry, meant to type

```
   when:
    - handle_cert_install | default(false) | bool
    - not (create_demo_certs | default(false) | bool)

```

---

Thanks @gene-redpanda. Now looks good.

---

Hey @gene-redpanda I have updated the code can you verify it and approve it.